lib/puppet/face/strings.rb

```diff
@@ -55,6 +56,9 @@ def check_required_features
       # all over the terminal. This should definitely not be in real code, but
       # it's very handy for debugging with pry
       #class YARD::Logger; def progress(*args); end; end
+      YARD::Tags::Library.define_directive("puppet.provider.param",
```

Comment:
Hmm... not sure how I feel about the name for this directive but I can't think of a better option off the top of my head. May be worth reaching out for suggestions.